src/main/webapp/js/all-section-deals.js

```diff
@@ -62,6 +77,61 @@ function initAllDeals() {
   }
 }
 
+/**
+ * Retrieves user's location
+ */
+function getLocation() {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(setNewURL, showError);
+  } else {
+    alert('Geolocation is not supported by this browser.');
```

Comment:
Let's just print some error to the console rather than showing an alert. Alerts feel spammy.